src/Compilers/CSharp/Portable/Syntax/SyntaxNodeRemover.cs

```diff
@@ -95,17 +95,21 @@ private void AddResidualTrivia(SyntaxTriviaList trivia, bool requiresNewLine = f
             {
                 if (requiresNewLine)
                 {
-                    this.AddEndOfLine();
+                    var eol = GetEndOfLine(trivia);
+                    if (eol.HasValue)
+                    {
+                        this.AddEndOfLine(eol.Value);
+                    }
```

Comment:
ths pattern seems to come up in like 4 different places in teh code.  can we extract helper?

Reply:
Now the check is moved into the AddEndOfLine method.